src/components/JSONFetcher/DataFetcher.js

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from "react";
+import ReactDOM from 'react-dom';
+import axios from 'axios';
+// import jsonData from '../../assets/JSON/event.JSON'
+
+const API='https://my-json-server.typicode.com/P3t3rP3r3z/JSONtestrepo';
+
+class DataFetcher extends Component {
+   constructor(props) {
+      super(props);
+      this.state = {
+        events: []
+      };
+    }
+ 
+    componentDidMount() {
+      fetch(API)
+        .then(response => response.json())
+        .then(
+          data => this.setState({events: data.events}));
+    }
+  
+    render() {
+      const { events } = this.state;
+      return(
+         <ul>
+            {events.map(event=>
+               <li key ={event.id}>
+               <a href={event.url}>{event.name}</a>
+               </li>
+            )}
+         </ul>
```

Comment:
To keep this generic we should keep this as a separate class with no jsx and just return an array of events that we can use to make jsx wherever. Or if this is the specific call for the events then it should be renamed to something else